fairscale/nn/misc/param_bucket.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Union
+
+import torch
+
+
+class ParamBucket:
```

Comment:
Why not just a Bucket class? Seem pretty similar to GradBuckets?

Reply:
see https://github.com/facebookresearch/fairscale/pull/540#issuecomment-803023686, there are differences still, I can make them inherit from the same base class if you'd like but don't expect huge line count savings. I'm being told that my PRs are too big, so probably better to keep that for another one ?

Reply:
I was only thinking in terms of making changes in two places. If you end up doing that you can rethink merging this. For now this looks great! very clean and easy to understand.